### DISCUSSION TITLE
Add array as new column

### DISCUSSION THREAD

**Discussion by landge** [uwdata/arquero#253] [Q&A] [answered]
Hi!

Is it possible to add an array (or adequate other data structure) as a new column to an existing table? 

In python/pandas you could do something like this:

`df = pandas.DataFrame()`
`df['column_name'] = [1,2,3]`

This would give you the following table:

||column_name|
|--|--|
|**0**|1|
|**1**|2|
|**2**|3|

How could this be done with arquero? Or do you have to use join to accomplish this?

Somehow I feel that I am missing something essential 😕 
Thanks for your help.

/Till

**Comment by jheer** [comment, marked as answer]
Yes, take a look at the [`table.assign()`](https://uwdata.github.io/arquero/api/table#assign) method. For example:

```js
const newTable = oldTable.assign({ column_name: [1, 2, 3] })
```

Of course, the new column lengths must match the old columns.

**Reply by cristianvogel** [reply]
is there something that can be done when the rowCount / column lengths don't match up?

**Comment by landge** [comment]
Thank you !! 🙏🏽

**Comment by cristianvogel** [comment]
This worked for me in Replit to add a column of indices to a table 
```oldTable.assign({ index: oldTable.indices() })```
but in context, the IDE gives a warning  `Argument type {index: Uint32Array} is not assignable to parameter type ColumnTable `
